buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt

```diff
@@ -216,88 +227,148 @@ fun enableOELPublishing(project: Project) {
                 else -> emptyList()
             }
 
-            extensions.getByType(PublishingExtension::class.java)
-                .publications.withType(DefaultMavenPublication::class.java)
-                // isAlias is needed for Gradle to ignore the fact that there's a
-                // publication that is not referenced as an available-at variant of the root module
-                // and has the Maven coordinates that are different from those of the root module
-                // FIXME: internal Gradle API! We would rather not create the publications,
-                //        but some API for that is needed in the Kotlin Gradle plugin
-                .all { publication ->
-                    if (publication.name == componentName) {
-                        publication.isAlias = true
-                    }
-                }
-
-            usages.forEach {    usage ->
+            usages.forEach { usage ->
+                // Use -published configuration because it would have correct attribute set
+                // required for publication.
                 val configurationName = usage.name + "-published"
 
-                configurations.matching{it.name == configurationName}.all() { conf ->
-                    conf.artifacts.clear()
-                    conf.dependencies.clear()
-                    conf.setExtendsFrom(emptyList())
-                    val composeVersion = requireNotNull(target.project.oelAndroidxVersion()) {
-                        "Please specify oel.androidx.version property"
-                    }
-                    val material3Version = requireNotNull(target.project.oelAndroidxMaterial3Version()) {
-                        "Please specify oel.androidx.material3.version property"
-                    }
-                    val foundationVersion = target.project.oelAndroidxFoundationVersion() ?: composeVersion
-                    val materialVersion = target.project.oelAndroidxMaterialVersion() ?: composeVersion
-
-                    val groupId = target.project.group.toString()
-                    val version = if (groupId.contains("org.jetbrains.compose.material3")) {
-                        material3Version
-                    } else if (groupId.contains("org.jetbrains.compose.foundation")) {
-                        foundationVersion
-                    } else if (groupId.contains("org.jetbrains.compose.material")){
-                        materialVersion
-                    } else {
-                        composeVersion
-                    }
-                    val newDependency = target.project.group.toString().replace("org.jetbrains.compose", "androidx.compose") + ":" + name + ":" + version
-                    conf.dependencies.add(target.project.dependencies.create(newDependency))
+                //
+                configurations.matching { it.name == configurationName }.all { conf ->
+                    newRootComponent.addUsageFromConfiguration(conf)
```

Comment:
Thanks!
I tested the local publication in one of the examples and it worked fine in both android and desktop. But I didn't look at the published files.
Different files size and hashes is surprising. I'll try to figure out what's going on.

Reply:
File sizes and hashes because it is built from another commit, with different sources, but with same build scripts.

I compared againg, with the neightbor commit - hash is the same:
<img width="1206" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/2403fee7-9d58-44e0-83ee-581f86339f80">

Reply:
Got it!
I'll be back with an answer about `ui-1.5.1-kotlin-tooling-metadata.json` and `the name field of the file is different in ui-1.5.1.module:`

Reply:
I've got the answers:

- `ui-1.5.1-kotlin-tooling-metadata.json` is needed for some statistics collection (not in JB) and was made for google purposes. It's not essential for the publication and we can proceed without it.
- `the name field of the file is different` - that's a minor thing too. It can break someone's project only when someone is not using gradle or have some very custom logic for dependency management. It should be possible to override the names too if such a need occurrs (if someone reports an issue, but they likely can  fix it in their custom solution too).